Src/CSharpier.Cli/Options/CaseInsensitiveEnumConverter.cs

```diff
@@ -0,0 +1,36 @@
+namespace CSharpier.Cli.Options;
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+internal class CaseInsensitiveEnumConverter<TEnum> : JsonConverter<TEnum>
+    where TEnum : struct
+{
+    public override TEnum Read(
+        ref Utf8JsonReader reader,
+        Type typeToConvert,
+        JsonSerializerOptions options
+    )
+    {
+        if (reader.TokenType != JsonTokenType.String)
+        {
+            throw new JsonException(
+                $"Expected a string for enum parsing, but got {reader.TokenType}."
+            );
+        }
+
+        var enumText = reader.GetString();
+        if (Enum.TryParse(enumText, true, out TEnum result))
```

Comment:
nit: booelan literal values are much more readable if you use named paramteters:
```suggestion
        if (Enum.TryParse(enumText, ignoreCase: true, out TEnum result))
```